### PR TITLE
Strong typing first round

### PR DESCRIPTION
**Pull request by schaefi**
Added strong typing for the following API methods
    
* logger
* logger_color_formatter
* logger_filter
* xml_description
* xml_state
* markup/any
* markup/base
* markup/xml
    
This references Issue #1644

### CODE REVIEW

**Comment by schaefi** [comment]
@davidcassany it took longer than I expected to get there. I know we agree to not change the behavior of the code but in one point I was not able to follow it.

In kiwi we supported reading the description as file (str) and as xml_content (bytes). All this in the same parameter. Of course this cannot be put into one type and was also bad practice. The reading from xml_content is not used by people and only tested in terms of the schema extension feature. It's easy to use a temporary file to handle raw content and so I decided to get rid of this concept and only allow reading a file

I hope this is ok and you will see it on review

**Comment by schaefi** [comment]
Looking on the codacy report next. This is going to be a hard ride but in the end the code will be better

**Comment by davidcassany** [comment]
>  I know we agreed to not change the behavior of the code but in one point I was not able to follow it.

Well, we new there was the change of facing some inconsistences. I think it is good if we sort those our even at the cost of slightly modifying the API. I'd say this is one of the motivations to make use of typing for the public API. So we make sure we have a consistent use of variables and also make the API explicit which is good for anyone using it.

File: kiwi/markup/base.py [hunk 1]
```diff
@@ -15,54 +15,39 @@
...
-                self.description = description
-            else:
-                self.description_temporary = NamedTemporaryFile()
-                with open(self.description_temporary.name, 'wb') as outfile:
-                    outfile.write(description)
-                self.description = self.description_temporary.name
-        except Exception as issue:
-            raise KiwiDescriptionInvalid(issue)
+    def __init__(self, description: str):
+        self.description = description
```
Comment (davidcassany) [inline kiwi/markup/base.py:34]: Makes sense this, I think it is a good idea to keep this description str variable as a single purpose. I see not much gain on keeping it just for extensions management. All in all the extensible XML feature is something to probably rethink, it turns to be quite complex extending the XML, I would even say far from obvious in a python virtual environment.

**Review by davidcassany — APPROVED** [review]
Looks good! I am glad we started doing so :+1:

**Pull request merged**